LanguageFeatures/Records/runtime_type_A01_t01.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The runtime type of a record is determined from the runtime types
+/// of its fields. There is no notion of a separate, explicitly reified type.
+/// So, here:
+/// ```dart
+/// (num, Object) pair = (1, 2.3);
+/// print(pair is (int, double)); // "true".
+/// ```
+/// The runtime type of `pair` is `(int, double)`, not `(num, Object)`. However,
+/// the variable declaration is still valid and sound because records are
+/// naturally covariant in their field types.
+///
+/// @description Checks that the runtime type of a record is determined from the
+/// runtime types of its fields
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+import "../../Utils/expect.dart";
+
+main() {
+  (num, Object) r1 = (1, 2.3);
+  Expect.runtimeIsType<(int, double)>(r1);
```

Comment:
This will verify that the run-time type of `r1` is a subtype of `(int, double)`, but it wouldn't reveal any problems if the run-time type were reported as, say, `(Never, Never)`.

The run-time types of built-in classes like `int` are somewhat tricky, because the implementations have different approaches: Native JIT based execution would have inlined SMIs, and it would use heap-allocated entities for large values, but both kinds of integer would report their type as `int` (and it's not just an instance of `Type` whose `toString()` is `'int'` in both cases, it's actually the same instance of `Type`).

So I think we _can_ actually test precisely, and we might want to do that as well (in addition to the current checks using `is`).

```dart
Type typeOf<X>() => X;

void main() {
  ...
  Expect.equals(typeOf<(int, double)>(), r1.runtimeType);
  ...
}
```

Reply:
Added